src/interp.rs

```diff
@@ -0,0 +1,143 @@
+        }
+    }
+
+}
+
+pub struct VM {
+    heap: Vec<Object>,
+    bytecode: Bytecode,
+    frames: Vec<Frame>,
+    pc: usize,
+}
+
+impl VM {
+    pub fn new(bytecode: Bytecode) -> VM {
+        VM {
+            heap: Vec::new(),
+            bytecode: bytecode,
+            frames: Vec::new(),
+            pc: 0,
+        }
+    }
+
+    pub fn run(&mut self) {
+        loop {
+            if self.pc >= self.bytecode.bytecode.len() {
+                break
+            }
+            let instr = self.bytecode.bytecode[self.pc].clone();
+            match instr {
+                Instr::PUSH_INT(ref x) => {
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.push(NativeType::Int(x.clone()));
+                    self.pc += 1
+                }
+                Instr::PUSH_STR(ref x) => {
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.push(NativeType::Str(x.clone()));
+                    self.pc += 1
+                }
+                Instr::POP => {
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.pop();
+                    self.pc +=1
+                }
+                Instr::ADD => panic!("NotYetImplemented"),
+                Instr::SUB => panic!("NotYetImplemented"),
+                Instr::LTEQ => panic!("NotYetImplemented"),
+                Instr::GTEQ => panic!("NotYetImplemented"),
+                Instr::LT => panic!("NotYetImplemented"),
+                Instr::GT => panic!("NotYetImplemented"),
+                Instr::EQEQ => panic!("NotYetImplemented"),
+                Instr::SWAP => panic!("NotYetImplemented"),
+                Instr::DUP => panic!("NotYetImplemented"),
+                Instr::LOAD_VAR(ref index) => {
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.load_local(index);
+                    self.pc += 1
+                }
+                Instr::STORE_VAR(ref name) => {
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.store_local(name);
+                    self.pc += 1
+                }
+                Instr::LOAD_GLOBAL(ref name) => panic!("NotYetImplemented"),
+                Instr::STORE_GLOBAL(ref name) => panic!("NotYetImplemented"),
+                Instr::NEW_OBJECT(ref class_name) => panic!("NotYetImplemented"),
+                Instr::LOAD_FIELD(ref field_name) => panic!("NotYetImplemented"),
+                Instr::STORE_FIELD(ref field_name) => panic!("NotYetImplemented"),
+                Instr::CALL(ref class_name, ref fn_name) => panic!("NotYetImplemented"),
+                Instr::JUMP_IF_TRUE(ref pos) => panic!("NotYetImplemented"),
+                Instr::JUMP_IF_FALSE(ref pos) => panic!("NotYetImplemented"),
+                Instr::JUMP(ref pos) => panic!("NotYetImplemented"),
+                Instr::RET =>  panic!("NotYetImplemented"),
+                _ => (),
```

Comment:
I think this clause shouldn't exist: it means that if you add a new instruction, the compiler won't warn you to add it in here too.